server/src/main/java/org/candlepin/model/OwnerInfoBuilder.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.model;
+
+import org.candlepin.dto.api.v1.ConsumptionTypeCountsDTO;
+import org.candlepin.dto.api.v1.OwnerInfo;
+
+import java.util.List;
+
+/**
+ * OwnerInfoBuilder is used to accumulate data to build OwnerInfo with dynamic values.
+ */
+
+public class OwnerInfoBuilder {
+
+    private OwnerInfo ownerInfo;
+
+    public static final String PHYSICAL = "physical";
+    public static final String GUEST = "guest";
+
+    public OwnerInfoBuilder() {
+        ownerInfo = new OwnerInfo();
```

Comment:
The original OwnerInfo constructor was doing some data initialization:
```
        consumerGuestCounts.put(GUEST, 0);
        consumerGuestCounts.put(PHYSICAL, 0);
```
We should also add that here, and add a unit test for it if one does not already exist

Reply:
Consumer count is set in `setConsumerGuestCounts` from `OwnerInfoCurator`. And this method is getting called before `info.build()`. Hence the Owner with no consumer is working as excepted. Still, I have added these into initialization as per the request.